src/panels/config/automation/condition/types/ha-automation-condition-sun.ts

```diff
@@ -22,111 +18,72 @@ export class HaSunCondition extends LitElement implements ConditionElement {
-        </ha-formfield>
-      </label>
-
-      <paper-input
-        .label=${this.hass.localize(
-          "ui.panel.config.automation.editor.conditions.type.sun.before_offset"
-        )}
-        name="before_offset"
-        .value=${before_offset}
-        @value-changed=${this._valueChanged}
-      ></paper-input>
-
-      <label>
-        ${this.hass.localize(
-          "ui.panel.config.automation.editor.conditions.type.sun.after"
-        )}
-
-        <ha-formfield
-          .label=${this.hass.localize(
+          ),
+        ],
+      ],
+    },
+    { name: "before_offset", selector: { text: {} } },
+    {
+      name: "after",
+      type: "select",
+      required: true,
+      options: [
+        [
+          "sunrise",
+          localize(
             "ui.panel.config.automation.editor.conditions.type.sun.sunrise"
-          )}
-        >
-          <ha-radio
-            name="after"
-            value="sunrise"
-            .checked=${after === "sunrise"}
-            @change=${this._radioGroupPicked}
-          ></ha-radio>
-        </ha-formfield>
-        <ha-formfield
-          .label=${this.hass.localize(
+          ),
+        ],
+        [
+          "sunset",
+          localize(
             "ui.panel.config.automation.editor.conditions.type.sun.sunset"
-          )}
-        >
-          <ha-radio
-            name="after"
-            value="sunset"
-            .checked=${after === "sunset"}
-            @change=${this._radioGroupPicked}
-          ></ha-radio>
-        </ha-formfield>
-      </label>
+          ),
+        ],
+      ],
+    },
+    { name: "after_offset", selector: { text: {} } },
+  ]);
 
-      <paper-input
-        .label=${this.hass.localize(
-          "ui.panel.config.automation.editor.conditions.type.sun.after_offset"
-        )}
-        name="after_offset"
-        .value=${after_offset}
+  protected render() {
+    const schema = this._schema(this.hass.localize);
+    return html`
+      <ha-form
+        .schema=${schema}
+        .data=${this.condition}
+        .hass=${this.hass}
+        .computeLabel=${this._computeLabelCallback}
         @value-changed=${this._valueChanged}
-      ></paper-input>
+      ></ha-form>
     `;
   }
 
   private _valueChanged(ev: CustomEvent): void {
-    handleChangeEvent(this, ev);
-  }
-
-  private _radioGroupPicked(ev: CustomEvent) {
-    const key = (ev.target as HaRadio).name;
     ev.stopPropagation();
-    fireEvent(this, "value-changed", {
-      value: {
-        ...this.condition,
-        [key]: (ev.target as HaRadio).value,
-      },
-    });
+    const newTrigger = ev.detail.value;
+    fireEvent(this, "value-changed", { value: newTrigger });
   }
 
-  static styles = css`
-    label {
-      display: flex;
-      align-items: center;
-    }
-  `;
+  private _computeLabelCallback = (schema: HaFormSchema): string =>
+    this.hass.localize(
+      `ui.panel.config.automation.editor.triggers.type.sun.${schema.name}`
```

Comment:
Should be condition? labels don't work now:

![image](https://user-images.githubusercontent.com/1444314/154742403-05825579-5614-441a-b50e-e04e0a231f8f.png)